src/flattenizer.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Flattens an object
+ *
+ * @param {Object} obj - the object to unflatten
+ * @returns {Object} - the flattened object, empty if provided object is undefined
+ */
+export const flatten = (obj) => {
+   let flattened = {};
+
+    for (let key in obj) {
+        if (!obj.hasOwnProperty(key))
+            continue;
+
+        if (typeof obj[key] === 'object') {
+            let flatObject = flatten(obj[key]);
+            for (let subKey in flatObject) {
+                // append to create new key value and assign it's value
+                flattened[key + '.' + subKey] = flatObject[subKey];
+            }
+        } else {
+            flattened[key] = obj[key];
+        }
+    }
+
+    return flattened;
+};
+
+/**
+ * Unflattens an object with compressed keys.
+ *
+ * @param {Object} flattened - object to unflatten
+ * @returns {Object} - the unflattened object, empty if provided object is undefined
+ */
+export const unflatten = flattened => {
+    let unflattened = {};
+
+    for (let prop in flattened) {
+        if (flattened.hasOwnProperty(prop)) {
+            explodeProperty(unflattened, prop, flattened);
+        }
+    }
+
+    return unflattened;
+};
+
+/**
+ * Explodes a flattened object key.
+ *
+ * @param {Object} currUnflattened the current unflattened object
+ * @param {String} prop the current property to explode
+ * @param {Object} flattenedObj the current flattened object
+ */
+const explodeProperty = (currUnflattened, prop, flattenedObj) => {
+    let keys = prop.split('.');
+    let value = flattenedObj[prop];
+    let lastKeyIndex = keys.length - 1;
+
+    for (let idx = 0; idx < lastKeyIndex; idx++) {
+        let currKey = keys[idx];
+
+        if (!(currKey in currUnflattened)) {
+            let nextKeyVal = parseInt(keys[idx + 1], 10);
```

Comment:
It seems that every non-number key has an array value, unless I'm missing something. Was that intentional? I guess I'm not sure why we need the `parseInt` logic here.